Week8/app/src/main/java/com/isabelledionisius/myomelettapp/worker/PeriodicWorker.kt

```diff
@@ -0,0 +1,56 @@
+package com.isabelledionisius.myomelettapp.worker
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import java.io.File
+import java.io.FileOutputStream
+import java.net.HttpURLConnection
+import java.net.URL
+
+class PeriodicWorker(context: Context, workerParameters: WorkerParameters) :
+    Worker(context,workerParameters ) {
+
+    // every worker has to override the doWork() function
+    // that describes what kind of work will run
+    // result can be a success or a failure
+    override fun doWork():Result {
+        // opening an http connection
+        val imageUrl = URL("http://www.recipepuppy.com/api/?i=onions,garlic&q=omelet")
```

Comment:
Why not use the RemoteApi here? You have all the things you need to build it here, so I'm not sure why you didn't just rely on that, and you instead chose to do things the manual way.